cmd/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/openshift/cluster-version-operator/pkg/cincinnati"
+	"github.com/openshift/cluster-version-operator/pkg/cvo"
+	"github.com/openshift/cluster-version-operator/pkg/version"
+
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+)
+
+const (
+	componentName = "cluster-version-operator"
+	leaseDuration = 90 * time.Second
+	renewDeadline = 45 * time.Second
+	retryPeriod   = 30 * time.Second
+)
+
+var (
+	flags struct {
+		kubeconfig string
+		clusterID  string
+		version bool
+	}
+)
+
+func init() {
+	flag.StringVar(&flags.kubeconfig, "kubeconfig", "", "Kubeconfig file to access a remote cluster. Warning: For testing only, do not use in production.")
+	flag.StringVar(&flags.clusterID, "cluster-id", "", "UUID of the cluster that the channel operator is managing, MUST be set")
```

Comment:
Should be part of a CVO config ?

Reply:
Eventually, yes. For this initial commit, I'm going to leave it as-is.